include/dlaf/permutations/general.h

```diff
@@ -148,7 +148,7 @@ void permute(comm::CommunicatorGrid grid, common::Pipeline<comm::Communicator>&
 /// This overload clones the row communicator (if Coord::Col) or column communicator (if Coord::Row) of
 /// @p grid internally.
 template <Backend B, Device D, class T, Coord coord>
-void permute(comm::CommunicatorGrid grid, SizeType i_begin, SizeType i_end,
+void permute(comm::CommunicatorGrid& grid, SizeType i_begin, SizeType i_end,
              Matrix<const SizeType, D>& perms, Matrix<const T, D>& mat_in, Matrix<T, D>& mat_out) {
   common::Pipeline<comm::Communicator> sub_task_chain(grid.subCommunicator(orthogonal(coord)).clone());
```

Comment:
To do: this should not use a bare `clone`. Also check other places for leftover `clone`s.

Reply:
This was the only `clone`. Updated to use the pipeline members of `COmmunicatorGrid`.